### ISSUE TITLE
random_string lookup returns random length string

### ISSUE THREAD

**Issue by clsmith70** [ansible-collections/community.general#6638]
### Summary

Our minimum password requirement dictates a minimum length of 14 characters.  We are using community.general.random_string to generate a random password with the length property set at 14.  I have noticed on occasion that the string returned is sometimes shorter than 14 characters.  In these cases, the account never enables due to password complexity requirements not being met.

### Issue Type

Bug Report

### Component Name

community.general.random_string

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.12.3]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.13 (default, Jun 14 2022, 17:49:07) [GCC 8.5.0 20210514 (Red Hat 8.5.0-13)]
  jinja version = 3.0.3
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general (from EE container)

# /usr/share/ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 6.5.0
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
(blank)
```

### OS / Environment

Red Hat Enterprise Linux release 8.7 (Ootpa) and Ansible Automation Platform 2.1.1-2

### Steps to Reproduce

```yaml (paste below)
- name: Define new user password
  no_log: true
  ansible.builtin.set_fact:
    gen_password: "{{ lookup('community.general.random_string', length=14, min_lower=1, min_upper=1, min_numeric=1, min_special=1, override_special='-_=+!#$()[]') }}"
```

### Expected Results

A string similar to `i31E_ax93mue[5`

### Actual Results

Sometimes a result like `e_5bD` might be returned.  Actual results cannot be provided due to no_log being True.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by clsmith70** [comment]
@Akasurde I will set something up in my lab and let you know the results.

**Comment by clsmith70** [comment]
@Akasurde I've run at least 100 iterations of a small template on a test server and always get a 14 character string.   I looked back through my job exports that include the temporary password set on the accounts using the same job template and role and find 4 different jobs out of 14 with a string value ranging from 3 to 10 characters, the rest all have 14 character strings.

The line used to generate that random string is exactly the same as the one I pasted above.  I also used that same line for the test I just ran.  The only way I might reproduce it is with a template defined in AAP, and see how many times I receive a string that is less than 14 characters in length.
